library/common/jni/jni_utility.cc

```diff
@@ -114,45 +114,51 @@ envoy_data* buffer_to_native_data_ptr(JNIEnv* env, jobject j_data) {
 }
 
 envoy_headers to_native_headers(JNIEnv* env, jobjectArray headers) {
+  return to_envoy_data(env, headers);
+}
+
+envoy_headers* to_native_headers_ptr(JNIEnv* env, jobjectArray headers) {
+  // Note: This check works for LocalRefs and GlobalRefs, but will not work for WeakGlobalRefs.
+  // Such usage would generally be inappropriate anyways; like C++ weak_ptrs, one should
+  // acquire a new strong reference before attempting to interact with an object held by
+  // a WeakGlobalRef. See:
+  // https://docs.oracle.com/javase/7/docs/technotes/guides/jni/spec/functions.html#weak
+  if (headers == NULL) {
+    return nullptr;
+  }
+
+  envoy_headers* native_headers = static_cast<envoy_headers*>(safe_malloc(sizeof(envoy_map_entry)));
+  *native_headers = to_native_headers(env, headers);
+  return native_headers;
+}
+
+envoy_stats_tags to_native_tags(JNIEnv* env, jobjectArray tags) { return to_envoy_data(env, tags); }
+
```

Comment:
this method is not new, it got marked as new bc i made changes to `to_native_headers`